test/e2e/e2e_test.go

```diff
@@ -1830,6 +1830,225 @@ var _ = ginkgo.Describe("e2e non-vxlan external gateway through a gateway pod",
+			if validIP == nil {
+				return false, nil
+			}
+			return true, nil
+		})
+		// Fail the test if no address is ever retrieved
+		if err != nil {
+			framework.Failf("Error trying to get the pod IP address")
+		}
+		// add a host route on the gateways for return traffic to the pod
+		_, err = runCommand("docker", "exec", gwContainer1, "ip", "route", "add", pingSrc, "via", nodeIP)
+		if err != nil {
+			framework.Failf("failed to add the pod host route on the test container %s: %v", gwContainer1, err)
+		}
+		_, err = runCommand("docker", "exec", gwContainer2, "ip", "route", "add", pingSrc, "via", nodeIP)
+		if err != nil {
+			framework.Failf("failed to add the pod host route on the test container %s: %v", gwContainer2, err)
+		}
+
+		// Verify the gateways and remote loopback addresses are reachable from the pod.
+		// Iterate checking connectivity to the loopbacks on the gateways until tcpdump see
+		// the traffic or 20 attempts fail. Odds of a false negative here is ~ (1/2)^20
+		ginkgo.By(fmt.Sprintf("Verifying ecmp connectivity to the external gateways by iterating through the prefix %s", exGWRemoteIpPrefix))
+
+		// create two channels that will handle error reporting from the goroutines
+		icmpChan1, icmpChan2 := make(chan error), make(chan error)
+
+		// Check for egress traffic to both gateway loopback addresses using tcpdump, since
+		// /proc/net/dev counters only record the ingress interface traffic is received on.
+		// The test will waits until an ICMP packet is matched on the gateways or fail the
+		// test if a packet to the loopback is not received within the timer interval.
+		// If an ICMP packet is never detected, return the error via the specified chanel.
+		go func() {
+			_, err = runCommand("docker", "exec", gwContainer1, "timeout", "20", "tcpdump", "-c", "1", "icmp")
+			if err != nil {
+				icmpChan1 <- err
+			}
+			framework.Logf("ICMP packet successfully detected on gateway %s", gwContainer1)
+			icmpChan1 <- err
+		}()
+		go func() {
+			_, err = runCommand("docker", "exec", gwContainer2, "timeout", "20", "tcpdump", "-c", "1", "icmp")
+			if err != nil {
+				icmpChan2 <- err
+			}
+			framework.Logf("ICMP packet successfully detected on gateway %s", gwContainer2)
+			icmpChan2 <- err
+		}()
+
+		// spawn a goroutine to asynchronously (to speed up the test)
+		// to ping the gateway loopbacks on both containers via ECMP.
+		for lastOctet := 1; lastOctet <= ecmpRetry; lastOctet++ {
+			gwLo := fmt.Sprintf("%s%d", exGWRemoteIpPrefix, lastOctet)
+			go func() {
+				_, err = framework.RunKubectl("exec", srcPingPodName, frameworkNsFlag, testContainerFlag, "--", "ping", "-w", "10", gwLo)
+				if err != nil {
+					framework.Logf("error generating a ping from the test pod %s: %v", srcPingPodName, err)
+				}
+			}()
+		}
+
+		// check for a test failure from the first gateway channel
+		if <-icmpChan1 != nil {
+			framework.Failf("failed to reach the mock gateway %s: %v", gwContainer1, err)
```

Comment:
if both gateway fail we only will know about the first one I think that something like this will work

```
err1 := <-icmpChan1
err2 := <-icmpChan2 

if err1 != nil || err2 != nil {
        framework.Failf("failed to reach the mock gateway &s: %v, %s: %v", gwContainer1, err1, gwContainer2, err2)
}
```

Reply:
SGTM, ty!